### DISCUSSION TITLE
Abstract Thesis

### DISCUSSION THREAD

**Discussion by polyglot-jones** [nbeversl/urtext#2] [Ideas]
(I'm posting this as an Issue, because you don't have Discussions turned on.)

Greetings. This project has very much **piqued my interest because I was planning on developing something along these lines myself** -- for journaling, research, prose writing, technical writing. I use **AsciiDoc** syntax for my work, so I planned on using a tiny subset of AsciiDoc for my note taking syntax. 

I see a lot of work here that _appears_ to be on point for me, and a lot of other work that I'm not sure really applies. (Just by mentioning AnyNode and RAKE, I know **we are both on the same wavelength**.) I'm certain that this development has been an exercise of thinking-out-loud and design-as-you-go (in a true agile spirit). **So, I was wondering if at this point you'd like to step back and discuss your findings so far?** A high-level description of the concepts and goals would be a great starting point, maybe with specific use-cases, to exemplify them. 

From there, I'd like to see if we could then pull out from your code **an abstract framework that is not dependent on any particular syntax**. Then I could inherit from the framework to apply my AsciiDoc syntax/semantics. 

**My Daily Journal Use-Case**

To give you an idea of where I'm coming from, I use SublimeText all day long to journal everything. I have a journal-entry snippet that's invoked with ctrl+j, ctrl+j.  It automatically generates a block ID (node ID) based on the datetime, as well as starting off the entry heading with the date/day.

    [[entry-2021-08-03-2206]]
    // 
    == 03 August 2021 Tuesday: 

Then, as I type in the title, the comment line above it is filled in with auto-generated hash-tags. For now, by the simple rule of making everything lower case. (And, now that I know about RAKE, I'll use it to make the hashtag generation smarter.)

    [[entry-2021-08-03-2206]]
    // #urtext #sublime-text
    == 03 August 2021 Tuesday: URTEXT for Sublime Text

    Urtext is a syntax and interpreter for plaintext.

    Uses:

    * prose writing
    * research
    * knowledge/information base
    * Zettelkasten

    The urtext code: https://github.com/nbeversl/urtext[]

    The urtext documentation (itself a urtext project): https://github.com/nbeversl/urtext-docs[]

    See also <<entry-2021-07-25-2144,SublimeText Package Development>>, <<entry-2021-01-02-2149,AsciiDoc Cheat Sheet>>.

The See-Also line I created by hand just to show you where I'm heading with this. For now, I just rely on Sublime Text's lightning fast multiple-file search to recall things.

My practice is that every thought/note begins life as a new journal entry. I just hit ^J^J and start typing. I call this "brain-dump" mode. Later, I'll massage the entry: delete it as irrelevant after all, rename it, move it to an earlier date if I'm documenting something that happened in the past, merge it with an earlier entry, etc.

**My Task-Management Use Case**

My aspirations include, for example, allowing an entry to contain a list of action items and have them automatically show up in my daily to-do list when appropriate, and then update the journal entry to show the action-item's status.

**Some Other Use Cases**

Printing (and re-printing) via PDF, certain journal entries for posterity.
Publishing certain journal entries to an EPUB to keep handy on my cell phone for reference.
Automatically process (classify, format, move) brain-dump entries based on context.

// Craig

**Comment by nbeversl** [comment]
Thank you for the interest. I've gone ahead and opened up discussions and moved this here. 

Have you had a chance to look at the documentation? It's in another repository at  https://github.com/nbeversl/urtext-docs. Though it is 4 months behind development at this point, it does offer an overview of the concepts and goals. We are working on bringing it up to date. In the meantime though, feel free to post specific questions here or in the documentation (where I've also opened up Discussions), and one of us will try to answer.

The code in this repository and at https://github.com/nbeversl/urtext_sublime (the Sublime Text plugin for the Desktop implementation) are both up to date (or at least only a week or so behind) development, so you can feel to try things out.
